.github/workflows/reusable-standard-its.yml

```diff
@@ -74,12 +74,15 @@ jobs:
         with:
           path: ~/.m2/repository
           key: maven-${{ runner.os }}-${{ inputs.build_jdk }}-${{ github.sha }}
-          fail-on-cache-miss: true
+
+      - name: Maven build
+        if: steps.maven-restore.outputs.cache-hit != 'true'
+        run: |
+          ./it.sh ci
```

Comment:
I wonder, why would the cache sometimes hit and sometimes not? Is the behavior non-deterministic? If so, is the cache atomic or might we sometimes get a partially-populated cache? Are we trying to populate the cache concurrently from reading from it?

This makes me wonder? What problem are we trying to solve? Do we know the root cause, or are we just adding "here's what to do it we get an unexpected result" workaround?

One really does hope that builds are deterministic!

Reply:
This is a known bug from GHA (https://github.com/actions/cache/issues/810). No, it's not atomic, so we're doing a clean build and running the tests as usual. As this is sporadic, failing the entire suite of tests due to a stuck cache didn't seem wise. Another workaround is to let that particular test with stuck cache download fail and rerun it after the workflow ends, hoping the problem doesn't occur again.

Reply:
Thanks for the explanation. It is surprising that caching isn't reliable. But, if that is indeed true, then the workaround is fine. Please add a comment that says basically what you explained above.

Reply:
That bug is unfortunate.  
This workaround will do great to unblock; but I'm going to do some tests with artifacts as a replacement for Caches that i was prototyping a bit ago.  The fact that if we cache miss means all the steps need to build again just sucks.  Caches are a problem for us, as Github only allows 10G before pruning, and that's basically one SHA's worth, so this may bite us hard when the project gets a lot of activity.